NDTensors/test/dense.jl

```diff
@@ -106,23 +108,23 @@ using Test
     @test Array(J * K) ≈ Array(J) * Array(K)
   end
 
-  @testset "Random constructor" begin
-    T = randomTensor((2, 2))
+  @testset "Random constructor" for op in ops
```

Comment:
Instead of having to define each `@testset` with a loop over `ops`, what about having a larger outer loop over `ops`? That would make it a bit more convenient for writing new tests.

Reply:
Does it matter that some tests that are not dependent on ops but will be in the ops loop?

Reply:
Good question. Maybe those could just be moved to a different part of the tests, outside of the outer loop over `ops`, so that they aren't run twice unnecessarily. That could also have the advantage that we can more easily track which tests are run for all devices and which ones are run only on CPU.